src/rgw/rgw_rest_s3.cc

```diff
@@ -6279,7 +6296,8 @@ int RGWSelectObj_ObjStore_S3::run_s3select(const char* query, const char* input,
 
   int header_size = 0;
 
-  if (m_s3_csv_object==0) {
+
+  if ((!m_parquet_type && m_s3_csv_object==0) || (m_parquet_type && m_s3_parquet_object==0)) {
```

Comment:
the clean and preferred syntax to test for a !null ptr is (!ptr) and conversely, (ptr) or (!!ptr), and unique_ptr has been designed to support that syntax.  It's no longer considered good style to write "0" (or even NULL) for nullptr, either, right?

Reply:
:+1: